### PR TITLE
iOS Wrapper: enabled HTML-based selectors such as `html>body`

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: css/amp.css [hunk 1]
```diff
@@ -59,7 +59,7 @@ html.-amp-ios-embed {
 }
 
 /** Wrapper for iOS-Embed-Wrapper mode. */
-i-amp-html-wrapper {
+#i-amp-html-wrapper {
```
Comment (aghassemi) [inline css/amp.css]: this is overridable now based on current validator CSS rules right? (AFAIK there is no restriction on id selectors) Maybe we need to switch to class='-amp-html-wrapper' to make it private.

File: src/service/viewport-impl.js [hunk 2]
```diff
@@ -1414,10 +1414,14 @@ export class ViewportBindingIosEmbedWrapper_ {
   constructor(win) {
     /** @const {!Window} */
     this.win = win;
+    const topClasses = this.win.document.documentElement.className;
+    this.win.document.documentElement.className = '';
     this.win.document.documentElement.classList.add('-amp-ios-embed');
 
     /** @private @const {!Element} */
-    this.wrapper_ = this.win.document.createElement('i-amp-html-wrapper');
+    this.wrapper_ = this.win.document.createElement('html');
```
Comment (aghassemi) [inline src/service/viewport-impl.js]: I am sure there is a good reason, but curious why just applying the CSS  to the existing <html> won't work and we must have a wrapper? ( two <html> elements is strange, I hope it doesn't trigger crazy edge-case bugs in Safari )

File: css/amp.css (hunk 1 above)
Comment (dvoytenko) [inline css/amp.css]: No, not overridable. `i-amp-` and `-amp-` prefixes are forbidden in selectors of any type.

File: src/service/viewport-impl.js (hunk 2 above)
Comment (dvoytenko) [inline src/service/viewport-impl.js]: Either way we have pretty horrible hacks and pretty horrible bugs. Through all the testing, however, this method appears to have fewer bugs and has a better performance. Among other things it allows header scrolling, flying carpet, correct scroll height calculation and many other benefits. Obviously, I can't wait for the day when we no longer need this.

File: css/amp.css (hunk 1 above)
Comment (aghassemi) [inline css/amp.css]: There might be  a validation bug then, #i-amp-html-wrapper passes validation on https://validator.ampproject.org/

Comment (dvoytenko) [inline css/amp.css]: /cc @Gregable @powdercloud Is this not blacklisted?

**Pull request merged**